lib/rules/no-deprecated-api.js

```diff
@@ -16,6 +16,18 @@ const getSemverRange = require("../util/get-semver-range")
 const extendTrackmapWithNodePrefix = require("../util/extend-trackmap-with-node-prefix")
 const unprefixNodeColon = require("../util/unprefix-node-colon")
 
+/**
+ * @typedef {Object} DeprecatedInfo
+ * @property {string} since
+ * @property {string|{ name: string, supported: string }[]|null} replacedBy
+ */
+/**
+ * @typedef {Object} ParsedOptions
```

Comment:
```suggestion
/**
 * @typedef DeprecatedInfo
 * @property {string} since
 * @property {string|{ name: string, supported: string }[]|null} replacedBy
 */
/**
 * @typedef ParsedOptions
```

When using `@property` no type is needed in `@typedef`, its implied that its an object